app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php

```diff
@@ -0,0 +1,227 @@
+
+/**
+ * Class Bolt_Boltpay_Block_Catalog_Product_Boltpay
+ *
+ * This block is used in boltpay/catalog/product/configure_checkout.phtml and boltpay/catalog/product/button.phtml templates
+ *
+ * This is defined in boltpay.xml config file
+ *
+ * The purpose is to add the Bolt to the Product Page,
+ * create the order on Bolt side through the javascript BoltCheckout.configureProductCheckout process.
+ *
+ */
+class Bolt_Boltpay_Block_Catalog_Product_Boltpay extends Mage_Core_Block_Template
+{
+    const CHECKOUT_TYPE_MULTI_PAGE  = 'multi-page';
+
+    /**
+     * Initiates sets up BoltCheckout config.
+     *
+     * @return string
+     */
+    public function getCartDataJsForProductPage()
+    {
+        try {
+            $currency = Mage::app()->getStore()->getCurrentCurrencyCode();
+
+            $productCheckoutCartItem = [];
+
+            $_product = Mage::registry('current_product');
+            if (!$_product) {
+                $msg = 'Bolt: Cannot find product info';
+                Mage::helper('boltpay/bugsnag')->notifyException($msg);
+
+                return '""';
+            }
+
+            $productCheckoutCartItem[] = [
+                'reference' => $_product->getId(),
+                'price'     => $_product->getPrice(),
+                'quantity'  => 1,
+                'image'     => $_product->getImageUrl(),
+                'name'  => $_product->getName(),
+            ];
+            $totalAmount = $_product->getPrice();
+
+            $productCheckoutCart = [
+                'currency' => $currency,
+                'items' => $productCheckoutCartItem,
+                'total' => $totalAmount,
+            ];
+
+             return json_encode($productCheckoutCart);
+
+        } catch (Exception $e) {
+            Mage::helper('boltpay/bugsnag')->notifyException($e);
+        }
+
+        return '""';
+    }
+
+    /**
+     * Collect callbacks for BoltCheckout.configureProductCheckout
+     *
+     * @param string $checkoutType
+     * @return string
+     */
+    public function getBoltCallbacks($checkoutType = self::CHECKOUT_TYPE_MULTI_PAGE)
+    {
+        /* @var Bolt_Boltpay_Helper_Api $boltHelper */
+        $boltHelper = Mage::helper('boltpay');
+
+        $checkCustom = $boltHelper->getPaymentBoltpayConfig('check', $checkoutType);
+        $onCheckoutStartCustom = $boltHelper->getPaymentBoltpayConfig('on_checkout_start', $checkoutType);
+        $onShippingDetailsCompleteCustom = $boltHelper->getPaymentBoltpayConfig('on_shipping_details_complete', $checkoutType);
+        $onShippingOptionsCompleteCustom = $boltHelper->getPaymentBoltpayConfig('on_shipping_options_complete', $checkoutType);
+        $onPaymentSubmitCustom = $boltHelper->getPaymentBoltpayConfig('on_payment_submit', $checkoutType);
+        $successCustom = $boltHelper->getPaymentBoltpayConfig('success', $checkoutType);
+        $closeCustom = $boltHelper->getPaymentBoltpayConfig('close', $checkoutType);
+
+        $onCheckCallback = '';
```

Comment:
suggest removing this if it's always empty string

Reply:
removed